hydrolib/core/io/ini/serializer.py

```diff
@@ -0,0 +1,246 @@
+        value = (
+            f"{property.value}{value_offset}"
+            if property.value is not None
+            else value_offset
+        )
+
+        comment = f" # {property.comment}" if property.comment is not None else ""
+
+        return [f"{indent}{key}{value}{comment}".rstrip()]
+
+    def _serialize_content(
+        self, content: Iterable[Union[Property, CommentBlock]], lengths: _Lengths
+    ) -> Iterable[str]:
+        def serialize_element(elem: Union[Property, CommentBlock]) -> Iterable[str]:
+            if isinstance(elem, Property):
+                return self._serialize_property(elem, lengths)
+            else:
+                return self._serialize_comment_block(
+                    elem, self._config.total_property_indent
+                )
+
+        return chain.from_iterable((serialize_element(elem) for elem in content))
+
+    def _serialize_datablock(
+        self, datablock: Optional[Iterable[Sequence[str]]], lengths: _Lengths
+    ) -> Iterable[str]:
+        if datablock is None or lengths.max_datablock_element_length is None:
+            return []
+
+        def serialize_row_element(elem: str, index: int) -> str:
+            offset = Serializer._get_offset(elem, lengths.max_datablock_element_length[index])  # type: ignore
+            return elem + offset
+
+        indent = " " * self._config.total_datablock_indent
+
+        def serialize_row(row: Sequence[str]) -> str:
+            return (
+                indent
+                + (" " * self._config.datablock_spacing)
+                .join((serialize_row_element(elem, i)) for elem, i in zip(row, count()))
+                .rstrip()
+            )
+
+        return (serialize_row(row) for row in datablock)
+
+    def _serialize_section(self, section: Section) -> Iterable[str]:
+        lengths = _Lengths.from_section(section)
+        header_iterable = self._serialize_section_header(section.header)
+        properties = self._serialize_content(section.content, lengths)
+        datablock = self._serialize_datablock(section.datablock, lengths)
+
+        return chain(
+            header_iterable,
+            properties,
+            datablock,
+        )
+
+    def serialize(self, document: Document) -> Iterable[str]:
+        """Serialize the provided document into an iterable of lines.
+
+        Args:
+            document (Document): The Document to serialize.
+
+        Returns:
+            Iterable[str]: An iterable returning each line of the serialized Document.
+        """
+        header_iterable = self._serialize_comment_header(document.header_comment)
+        sections_iterable = chain.from_iterable(
+            Serializer._interweave(
+                (self._serialize_section(section) for section in document.sections),
+                [""],
+            )
+        )
+        return chain(header_iterable, sections_iterable)
+
+
+def write_ini(
+    path: Path, document: Document, config: Optional[SerializerConfig] = None
```

Comment:
```suggestion
    path: Path, document: Document, config: Optional[SerializerConfig] = SerializerConfig()
```

Reply:
Or is it because you are checking for None anyway?

Reply:
It is generally considered bad practice to use mutable objects as default values for calls, because this single default value instance is used among all calls. This would mean that if the call that uses this default value would make changes to the state, each subsequent call would have these changes as well. As such, the general approach is to set the default value to None, and then checking if it is, and creating the default value. 

 In this case it could work to use SerializerConfig directly because the code should not be changing it. However for the sake of consistency I opted to use None here as well.

Reply:
Thanks for the explanation!